libgeopmd/test/PowercapSysfsDriverTest.cpp

```diff
@@ -0,0 +1,200 @@
+        geopm::Exception
+    ) << "Should throw an exception for an invalid signal name";
+}
+
+TEST_F(PowercapSysfsDriverTest, attribute_path_valid_and_invalid_cases)
+{
+    std::string valid_path = m_driver->attribute_path("POWERCAP::CPU_POWER_LIMIT", 0);
+    EXPECT_FALSE(valid_path.empty()) << "Attribute path for a valid signal should not be empty";
+
+    EXPECT_THROW(
+        m_driver->attribute_path("INVALID_SIGNAL", 0),
+        geopm::Exception
+    ) << "Should throw an exception for an invalid signal name";
+
+    EXPECT_THROW(
+        m_driver->attribute_path("POWERCAP::CPU_POWER_LIMIT", -1),
+        geopm::Exception
+    ) << "Should throw an exception for an invalid domain index";
+}
+
+TEST_F(PowercapSysfsDriverTest, signal_parse_valid_and_invalid_cases)
+{
+    auto parse_function = m_driver->signal_parse("POWERCAP::CPU_POWER_LIMIT");
+
+    EXPECT_DOUBLE_EQ(1.5, parse_function("1500000"))
+        << "Signal parse should correctly scale the value";
+
+    EXPECT_TRUE(std::isnan(parse_function("INVALID_VALUE")))
+        << "Signal parse should return NaN for invalid input";
+
+    EXPECT_TRUE(std::isnan(parse_function("")))
+        << "Signal parse should return NaN for empty input";
+
+    EXPECT_THROW(
+        m_driver->signal_parse("INVALID_SIGNAL"),
+        geopm::Exception
+    ) << "Should throw an exception for an invalid signal name";
+}
+
+TEST_F(PowercapSysfsDriverTest, control_gen_valid_and_invalid_cases)
+{
+    auto control_function = m_driver->control_gen("POWERCAP::CPU_POWER_LIMIT");
+
+    EXPECT_EQ("1500000", control_function(1.5))
+        << "Control generation should correctly scale the value";
+
+    EXPECT_EQ("0", control_function(0.0))
+        << "Control generation should handle zero value correctly";
+
+    EXPECT_EQ("-1500000", control_function(-1.5))
+        << "Control generation should handle negative values correctly";
+
+    EXPECT_THROW(
+        m_driver->control_gen("INVALID_CONTROL"),
+        geopm::Exception
+    ) << "Should throw an exception for an invalid control name";
+}
+
+TEST_F(PowercapSysfsDriverTest, properties_are_loaded_correctly)
+{
+    auto properties = m_driver->properties();
+    EXPECT_FALSE(properties.empty()) << "Properties should not be empty";
+
+    for (const auto &property : properties) {
+        EXPECT_FALSE(property.first.empty()) << "Property name should not be empty";
+        EXPECT_FALSE(property.second.attribute.empty()) << "Attribute should not be empty for property: " << property.first;
+        EXPECT_GT(property.second.scaling_factor, 0.0) << "Scaling factor should be positive for property: " << property.first;
+    }
+}
+
+TEST_F(PowercapSysfsDriverTest, driver_and_plugin_name_match)
+{
+    EXPECT_EQ("POWERCAP", m_driver->driver())
+        << "Driver name should match the plugin name";
+    EXPECT_EQ("POWERCAP", PowercapSysfsDriver::plugin_name())
+        << "Plugin name should be POWERCAP";
+}
```

Comment:
It seems this is a duplicate of the first test (`iogroup_plugin_name_matches_driver_name`).